node_modules/oae-principals/lib/api.user.js

```diff
@@ -723,3 +961,109 @@ var _setAdmin = function(ctx, adminType, isAdmin, principalId, callback) {
         return PrincipalsDAO.setAdmin(adminType, isAdmin, principalId, callback);
     });
 };
+
+/**
+ * Send an email token to a user
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {User}       user            The user to send the email token to
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+var sendEmailToken = module.exports.sendEmailToken = function(ctx, user, callback) {
+    callback = callback || function(err) {
+        if (err) {
+            log().error({'err': err, 'userId': user.id}, 'Unable to send a user an email verification email');
+        }
+    };
+
+    // Generate a token that will be sent to the user's email address
+    var token = _generateEmailToken();
+
+    // Store the token
+    PrincipalsDAO.storeEmailToken(token, user.id, function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        // Send an email to the specified e-mail address
+        var data = {
+            'tenant': ctx.tenant(),
+            'user': user,
+            'baseUrl': TenantsUtil.getBaseUrl(ctx.tenant()),
+            'skinVariables': require('oae-ui').getTenantSkinVariables(ctx.tenant().alias),
+            'token': token
+        };
+        EmailAPI.sendEmail('oae-principals', 'verify', user, data, {'hash': token, 'ignoreVerification': true}, callback);
+    });
+};
+
+/**
+ * Resend an email token
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {String}     userId          The id of the user for who to resend the email token
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+var resendEmailToken = module.exports.resendEmailToken = function(ctx, userId, callback) {
+    var validator = new Validator();
+    validator.check(userId, {'code': 400, 'msg': 'A valid user id must be provided'}).isUserId();
+    validator.check(null, {'code': 401, 'msg': 'You have to be logged in to be able to resend an email token'}).isLoggedInUser(ctx);
```

Comment:
I'm not entirely how this works. Wouldn't you only be able to trigger a resend for the current user (unless you're an admin)?